tank_battalion/enemy.pde

```diff
@@ -9,13 +9,19 @@ class Enemy {
   float rotate_timer = 0.0f;
   float rotate_delay = 0.0f;
   int direction = 1;
+  boolean rainbow ;
+  color tint =color(255,0,0);
+  float tint_cooldown = 0.2f;
+  color color_blue =color(0,0,255);
+  color color_red =color(255,0,0);
 
   final String SPRITES_FOLDER = "../assets/sprites/";
   PImage enemy_up, enemy_down, enemy_left, enemy_right, actual_image;
   
-  Enemy(int xpos, int ypos){
+  Enemy(int xpos,boolean Rainboww, int ypos){
     x = xpos;
     y = ypos;
+    rainbow = true; 
```

Comment:
You forgot to set this back to `rainbow = Rainboww;`. With this code every tank will be a rainbow tank.